app/Domains/Vault/ManageVault/Api/Controllers/VaultController.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace App\Domains\Vault\ManageVault\Api\Controllers;
+
+use App\Domains\Vault\ManageVault\Services\CreateVault;
+use App\Domains\Vault\ManageVault\Services\DestroyVault;
+use App\Domains\Vault\ManageVault\Services\UpdateVault;
+use App\Http\Controllers\ApiController;
+use App\Http\Resources\VaultResource;
+use App\Models\Vault;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Database\QueryException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class VaultController extends ApiController
+{
+    public function __construct()
+    {
+        $this->middleware('abilities:read')->only(['index', 'show']);
+        $this->middleware('abilities:create')->only(['store']);
+        $this->middleware('abilities:update')->only(['update']);
+        $this->middleware('abilities:destroy')->only(['delete']);
+
+        parent::__construct();
+    }
+
+    /**
+     * List all vaults
+     *
+     * Get all the vaults in the account.
+     *
+     * @group Vault management
+     * @subgroup Vaults
+     * @queryParam limit int A limit on the number of objects to be returned. Limit can range between 1 and 100, and the default is 10. Example: 10
+     * @apiResourceModel App\Models\Vault
+     */
```

Comment:
Lets try with the attributes:
```suggestion
     */
    #[QueryParam('limit', 'int', description: 'A limit on the number of objects to be returned. Limit can range between 1 and 100, and the default is 10.', required: false, example: 10)]
    #[ResponseFromApiResource(VaultResource::class, Vault::class, collection: true)]
```